### PR TITLE
access_control_request_headers empty set always valid

### PR DESCRIPTION
**Pull request by MichaelSitter**
Coveralls says this codepath is already covered?

### CODE REVIEW

**Comment by MichaelSitter** [comment]
ping @jopecko

**Comment by relistan** [comment]
@MichaelSitter covered doesn't mean that this edge case is tested. Please add a test for empty set.

File: lib/spacesuit/cors_middleware.ex [hunk 1]
```diff
@@ -139,7 +139,11 @@ defmodule Spacesuit.CorsMiddleware do
   end
 
   defp valid_control_headers?(headers) do
-    MapSet.subset?(headers, access_control_request_headers())
+    if MapSet.size(headers) == 0 do
```
Comment (relistan) [inline lib/spacesuit/cors_middleware.ex]: The body of this whole method can just read:

```elixir
MapSet.size(headers) == 0 || MapSet.subset?(headers, access_control_request_headers())
```

You can get rid of the whole `if`/`else` block

**Comment by MichaelSitter** [comment]
Added test, also found a case in preflight max age which wasn't covered.
```
COV    FILE                                        LINES RELEVANT   MISSED
 70.4% lib/mix/tasks/validate_routes.ex               76       27        8
100.0% lib/spacesuit.ex                               32        4        0
100.0% lib/spacesuit/api_message.ex                   17        2        0
 83.3% lib/spacesuit/auth_middleware.ex               35       12        2
100.0% lib/spacesuit/cors_middleware.ex              246       64        0
100.0% lib/spacesuit/debug_middleware.ex              12        3        0
  0.0% lib/spacesuit/health_handler.ex                18        3        3
 50.0% lib/spacesuit/http_client.ex                   50        4        2
 31.3% lib/spacesuit/http_server.ex                   96       16       11
 44.0% lib/spacesuit/proxy_handler.ex                168       50       28
100.0% lib/spacesuit/router.ex                       153       42        0
 82.9% lib/spacesuit/session_service.ex              171       41        7
100.0% lib/spacesuit/supervisor.ex                    18        3        0
[TOTAL]  77.5%
```

**Comment by relistan** [comment]
Thanks @MichaelSitter!

File: lib/spacesuit/cors_middleware.ex [hunk 2]
```diff
@@ -139,7 +139,7 @@ defmodule Spacesuit.CorsMiddleware do
   end
 
   defp valid_control_headers?(headers) do
-    MapSet.subset?(headers, access_control_request_headers())
+    MapSet.size(access_control_request_headers()) == 0 || MapSet.subset?(headers, access_control_request_headers())
```
Comment (jopecko) [inline lib/spacesuit/cors_middleware.ex:142]: sorry I'm looking at this after merged, but an optimization for a future patch; in case where there are configured headers, this calls [same logic twice](https://github.com/relistan/spacesuit/blob/master/lib/spacesuit/cors_middleware.ex#L240). I.e.,
```
defp access_control_request_headers do
    (cors()[:access_control_request_headers] || [])
    |> Enum.map(&String.downcase/1)
    |> Enum.into(MapSet.new)
  end
```
